docs/appendix.md

```diff
@@ -34,6 +34,74 @@ In the middle column, a client-side-only render, we wrap the component in an Isl
 
 In the right-most column, a server-side render with client hydration, we wrap the component in an Island without the `clientOnly` option set. Immediately the client will display the server-side rendered “Hello, world!”, and then the Island’s React code will be downloaded and [instantiated](https://react.dev/reference/react-dom/client/hydrateRoot) (“hydrated”), with the first render also returning “Hello, world!”, since `useAfterIslandHydration` will return `false` on its first invocation. Automatically, there is a subsequent render where `useAfterIslandHydration` will resolve to `true` and the client will show “Hello, hydration!”.
 
+## Debugging
+
+Client and Server side rendering means that html is being rendered in two different contexts. With "Islands", React components are generally executed in _both_ places. Often times a useful tool is logging various values during the render. However, while a `console.log` statement that executes during a server render will log, the server render logs are not visible to the developer. `@hubspot/cms-components` exposes some useful utilities for logging values on the server which then get shipped to the client.
```

Comment:
I think the main thing to highlight here is what this means for server rendering - something not in an island has no opportunity ever to log in the browser without these utils

```suggestion
Logging various values in the render lifecycle of a component is a useful strategy for debugging React components. However, if a component is rendered only on the server, a `console.log()` result is currently not visible to the developer when deployed. It may also be useful to compare values in components within an island, which are rendered both on the server and the client. `@hubspot/cms-components` exposes some useful utilities for logging values on the server which then get shipped to the client to visibly log in the browser.
```